plugins/kinetic-dp/fu-kinetic-dp-device.c

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (C) 2015 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2016 Mario Limonciello <mario.limonciello@dell.com>
+ * Copyright (C) 2021 Jeffrey Lin <jlin@kinet-ic.com>
+ * Copyright (C) 2022 Hai Su <hsu@kinet-ic.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fcntl.h>
+
+#include "fu-kinetic-dp-common.h"
+#include "fu-kinetic-dp-connection.h"
+#include "fu-kinetic-dp-device.h"
+#include "fu-kinetic-dp-firmware.h"
+#include "fu-kinetic-dp-puma-aux-isp.h"
+#include "fu-kinetic-dp-secure-aux-isp.h"
+
+struct _FuKineticDpDevice {
+	FuUdevDevice parent_instance;
+	FuKineticDpAuxIsp *aux_isp_ctrl;
+	gchar *system_type;
+	FuKineticDpFamily family;
+	FuKineticDpMode mode;
+};
+
+G_DEFINE_TYPE(FuKineticDpDevice, fu_kinetic_dp_device, FU_TYPE_UDEV_DEVICE)
+
+FuKineticDpAuxIsp *
+fu_kinetic_dp_device_get_aux_isp_ctrl(FuKineticDpDevice *self)
+{
+	g_debug("device get aux isp ctrl.");
+	g_return_val_if_fail(FU_IS_DEVICE(self), NULL);
+	return self->aux_isp_ctrl;
+}
+
+/* hookup isp protocol for different chip */
+void
+fu_kinetic_dp_device_set_aux_isp_ctrl(FuKineticDpDevice *self, KtChipId chip_id)
+{
+	g_return_if_fail(FU_IS_KINETIC_DP_DEVICE(self));
+	g_return_if_fail(self->aux_isp_ctrl == NULL);
+
+	if (chip_id == KT_CHIP_JAGUAR_5000 || chip_id == KT_CHIP_MUSTANG_5200) {
+		g_set_object(&self->aux_isp_ctrl,
+			     FU_KINETIC_DP_AUX_ISP(fu_kinetic_dp_secure_aux_isp_new()));
+		g_debug("device set aux isp ctrl for Jaguar or Mustang.");
+	} else if (chip_id == KT_CHIP_PUMA_2900) {
+		g_set_object(&self->aux_isp_ctrl,
+			     FU_KINETIC_DP_AUX_ISP(fu_kinetic_dp_puma_aux_isp_new()));
+		g_debug("device set aux isp ctrl for Puma.");
+	}
+}
+
+void
+fu_kinetic_dp_device_set_system_type(FuKineticDpDevice *self, const gchar *system_type)
+{
+	g_return_if_fail(FU_IS_KINETIC_DP_DEVICE(self));
+	self->system_type = g_strdup(system_type);
+	g_debug("device system type set %s", system_type);
```

Comment:
i think this belongs in `->to_string()` rather than as a g_debug.

Reply:
fixed